### PR TITLE
Convert To Interpolated Strings refactoring provider should treat cha…

### PR DESCRIPTION
**Pull request by YairHalberstadt**
…r literals the same way it treats string literals.

See https://github.com/dotnet/roslyn/issues/37324

### CODE REVIEW

File: src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs [hunk 1]
```diff
@@ -113,12 +115,13 @@ private Task<Document> UpdateDocumentAsync(Document document, SyntaxNode root, S
...
-                var currentContentIsStringLiteral = syntaxFacts.IsStringLiteralExpression(piece);
-                if (currentContentIsStringLiteral)
+                var isCharacterLiteral = syntaxFacts.IsCharacterLiteralExpression(piece);
+                var currentContentIsStringOrCharacterLiteral = syntaxFacts.IsStringLiteralExpression(piece) || isCharacterLiteral;
+                if (currentContentIsStringOrCharacterLiteral)
                 {
                     var text = piece.GetFirstToken().Text;
                     var textWithEscapedBraces = text.Replace("{", "{{").Replace("}", "}}");
-                    var textWithoutQuotes = GetTextWithoutQuotes(textWithEscapedBraces, isVerbatimStringLiteral);
+                    var textWithoutQuotes = GetTextWithoutQuotes(textWithEscapedBraces, isVerbatimStringLiteral, isCharacterLiteral);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs]: dumb question. how does this work if you have a normal char lit with something like `\u000a` and you have that in either an `$""` interpolated string or an `@$""` interpolated string?  Presumably we need to do some escaping for one or hte other.

Comment (YairHalberstadt) [reply src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs]: We don't interpolate verbatim strings with non verbatim

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs]: That is the case with strings.  But you're adding support for char literals (which have no verbatim/non-verbatim split).  How are you ensuring the char literal form matches the string form?

Comment (YairHalberstadt) [reply src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs]: Line 64 we add char literals to the list of string literals. Later on we check that if any are verbatim, all are verbatim.

Comment (YairHalberstadt) [reply src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs]: So you simply cannot concatenate a char with a verbatim string. I think this is fair enough, as the same would be true for a one letter non verbatim string, whatever it's contents.

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs]: got it.  do we have a test for this?  Thanks!

Comment (YairHalberstadt) [reply src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs]: Yes TestMissingWithMixedStringTypes3 in ConvertConcatenatedToInterpolatedStringTests.cs

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs]: 👍

**Comment by JoeRobich** [comment]
@YairHalberstadt Let me know if you are good with this PR, and we'll get it in. Thanks!

**Comment by YairHalberstadt** [comment]
@JoeRobich
Other than that missing +. However I'm on holiday and won't be able to fix it for another couple of weeks. If you prefer, feel free to fix it yourself, else I'll do it when I get the chance.

**Comment by YairHalberstadt** [comment]
The  missing plus has been fixed, so this should be good to merge @JoeRobich

**Comment by JoeRobich** [comment]
Thanks Yair! I'll kick off the timed out integration test and we'll get it merged in.

**Pull request merged**